### PR TITLE
Release configuration using github actions

### LINKED ISSUE #24: Automatic release: add and test action for it
There are several tools in the market place:
-- https://github.com/actions/create-release
-- https://github.com/marketplace/actions/maven-release

### PR DESCRIPTION
**Pull request by petertrr**
This closes #24 

### What's done:
* Workflow which is triggered by push of tag into master and uploads artifacts to artipie

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -215,4 +232,38 @@
...
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>de.jutzig</groupId>
+                        <artifactId>github-release-plugin</artifactId>
+                        <version>1.4.0</version>
+                        <configuration>
+                            <description>Description of your release</description>
+                            <releaseName>1.0 Final</releaseName>
+                            <tag>${project.version}</tag>
```
Comment (akuleshov7) [inline pom.xml:247]: this will create tag from the version installed in pom.xml?

Comment (petertrr) [reply pom.xml:247]: Whoa, this appeared here by accident, I eventually decided not to use this plugin.

Comment (petertrr) [reply pom.xml:247]: Tag is should be created manually with `git tag`

File: pom.xml [hunk 2]
```diff
@@ -4,13 +4,12 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.cqfn.diktat</groupId>
     <artifactId>diktat-parent</artifactId>
-    <version>${diktat.version}</version>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (akuleshov7) [inline pom.xml:7]: it will be updated automatically?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Deploy artifacts
+        run: mvn -B clean deploy -Dusername=${{ secrets.ARTIPIE_USER }} -Dpassword=${{ secrets.ARTIPIE_PASSWORD }}
+      - name: Upload assets to github releases
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./diktat-rules/target/diktat-rules-${{ env.RELEASE_VERSION }}-jar-with-dependencies.jar
```
Comment (akuleshov7) [inline .github/workflows/release.yml:45]: don't you know - if there is any chance to rename 'jar-with-dependencies' suffix?

Comment (akuleshov7) [reply .github/workflows/release.yml:45]: I googled it -                         <descriptorRef>jar-with-dependencies</descriptorRef>
does not work

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          asset_name: diktat-rules-${{ env.RELEASE_VERSION }}-jar-with-dependencies.jar
+          asset_content_type: application/zip
+      - name: Increment development version
+        run: |
+          mvn -B versions:set -DnextSnapshot=true -DprocessAllModules=true versions:commit
+          git add *pom.xml
```
Comment (akuleshov7) [inline .github/workflows/release.yml:51]: git add?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          git config --local user.email "diktat@cqfn.org"
+          git config --local user.name "diktat_build"
```
Comment (akuleshov7) [inline .github/workflows/release.yml:53]: are you sure with this e-mail?

File: pom.xml (hunk 2 above)
Comment (petertrr) [reply pom.xml:7]: It will be updated by `mvn versions:set` in workflow, all modules at once.

File: .github/workflows/release.yml (hunk 4 above)
Comment (petertrr) [reply .github/workflows/release.yml:51]: Well, the command before this sets versions to new, `versions:commit` has nothing to do with VCS, it just removes backup pom if they have been created by this plugin. So we need to commit updated poms manually.

File: .github/workflows/release.yml (hunk 5 above)
Comment (petertrr) [reply .github/workflows/release.yml:53]: No, this is the first one that came to my mind. But we should put either one of our own emails or use some fake email for bot. Do you have other suggestions?

File: .github/workflows/release.yml (hunk 3 above)
Comment (petertrr) [reply .github/workflows/release.yml:45]: I guess,  `finalName` and `appendAssemblyId` in maven-assembly-plugin can let us rename the archive.

**Pull request merged**